### PR TITLE
HDDS-1191. Replace Ozone Rest client with S3 client in smoketests and docs

### PR DESCRIPTION
**Pull request by elek**
As it's discussed in the the parent jira the rest support for Ozone Client protocol can be removed to use S3 Rest API instead of that.

Some of the unit tests are already disabled, so it seems to be better to remove it from the documentation (and from the smoketests).

See: https://issues.apache.org/jira/browse/HDDS-1191

### CODE REVIEW

File: hadoop-hdds/docs/content/CommandShell.md [hunk 1]
```diff
@@ -71,15 +71,8 @@ _\[scheme\]\[server:port\]/volume/bucket/key_
...
-1. Scheme - Can be one of the following
-    * o3  - Ozone's native RPC protocol. If you specify this scheme, the
-    native RPC protocol is used while communicating with Ozone Manager and
-    data nodes.
-    * http/https - If an HTTP protocol is specified, then Ozone shell assumes
-     that you are interested in using the Ozone Rest protocol and falls back
-     to using the REST protocol instead of RPC.
- If no protocol is specified, the Ozone shell defaults to the native RPC
- protocol.
+1. Scheme - This should be `o3` which is the native RPC protocol to access
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/CommandShell.md]: whitespace:end of line

File: hadoop-hdds/docs/content/JavaApi.md [hunk 2]
```diff
@@ -25,31 +25,19 @@ menu:
 Introduction
 -------------
 
-Ozone ships with its own client library that supports both RPC (Remote
-Procedure Call) and REST (Representational State Transfer). This library is
-the primary user interface to ozone.
+Ozone ships with its own client library that supports RPC. For generic use cases the S3
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/JavaApi.md]: whitespace:end of line

File: hadoop-hdds/docs/content/JavaApi.md [hunk 3]
```diff
@@ -25,31 +25,19 @@ menu:
 Introduction
 -------------
 
-Ozone ships with its own client library that supports both RPC (Remote
-Procedure Call) and REST (Representational State Transfer). This library is
-the primary user interface to ozone.
+Ozone ships with its own client library that supports RPC. For generic use cases the S3 
+compatible REST interface also can be used instead of the Ozone client.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/JavaApi.md]: whitespace:end of line

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 36 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 58 | Maven dependency ordering for branch |
| +1 | mvninstall | 1088 | trunk passed |
| +1 | compile | 928 | trunk passed |
| -1 | mvnsite | 113 | dist in trunk failed. |
| +1 | shadedclient | 2921 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 46 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 24 | Maven dependency ordering for patch |
| -1 | mvninstall | 17 | dist in the patch failed. |
| +1 | compile | 902 | the patch passed |
| +1 | javac | 902 | the patch passed |
| -1 | mvnsite | 28 | dist in the patch failed. |
| -1 | whitespace | 0 | The patch has 3 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | shadedclient | 599 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 43 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 26 | docs in the patch passed. |
| -1 | unit | 28 | dist in the patch failed. |
| +1 | asflicense | 38 | The patch does not generate ASF License warnings. |
| | | 4924 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-529/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/529 |
| Optional Tests |  dupname  asflicense  mvnsite  compile  javac  javadoc  mvninstall  unit  shadedclient  |
| uname | Linux 07ed38007244 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 538bb48 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-529/1/artifact/out/branch-mvnsite-hadoop-ozone_dist.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-529/1/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-529/1/artifact/out/patch-mvnsite-hadoop-ozone_dist.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-529/1/artifact/out/whitespace-eol.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-529/1/artifact/out/patch-unit-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-529/1/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs hadoop-ozone/dist U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-529/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by anuengineer** [comment]
:+1:  I will commit this soon.

**Pull request closed without merging**